### PR TITLE
Fix weird bug

### PR DESCRIPTION
**Pull request by nnistelrooij**
the model is now able to train again. I had to add `random_state=2020` when splitting the data.

Also added a on-plateau learning rate scheduler.

Also save the best-performing model to storage.

Also let x-axis of TensorBoard be number of unique training images the model has been trained on.

Also cleaned up the `optim.py` file by separating the training, validation and epoch loops.

### CODE REVIEW

**Comment by rusane** [comment]
Just some clarification on the issue: the `random_state` is necessary to keep the training/validation images and their corresponding labels paired (i.e., use the same split on the images and the labels),  that's why it was there before already. The poor performance can then be explained because the images were paired with incorrect labels (for both train and validation set).

**Comment by nnistelrooij** [comment]
You can split the images and labels with one call to ` train_test_split`. Then you don't need the `random_state` parameter.

Can you add that functionality before merging?

**Comment by rusane** [comment]
Sure, I'll do it in a moment. The `random_state` does introduce consistency (always the same split) though, i.e, comparing the performance of different model runs would be fairer if we used the same split.

**Comment by rusane** [comment]
I left the `random_state` parameter in the function call, but if you have a good reason to remove it, be my guest.

**Comment by nnistelrooij** [comment]
Best of both worlds! I added the seed of data splitting as a console argument.
Are you happy to merge?
